fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -739,7 +745,30 @@ public FedoraResource getUnfrozenResource() {
         }
 
         try {
-            return new FedoraResourceImpl(getSession().getNodeByIdentifier(getProperty("jcr:frozenUuid").getString()));
+            // Either this resource is frozen
+            if (hasProperty(JCR_FROZEN_UUID)) {
+                try {
+                    return new FedoraResourceImpl(getNodeByProperty(getProperty(JCR_FROZEN_UUID)));
+                } catch (ItemNotFoundException e) {
+                    // The unfrozen resource has been deleted, return the tombstone.
+                    return new TombstoneImpl(getNode());
+                }
+
+                // ..Or it is a child-version-history on a frozen path
+            } else if (hasProperty(JCR_CHILD_VERSION_HISTORY)) {
+                final Node childVersionHistory = getNodeByProperty(getProperty(JCR_CHILD_VERSION_HISTORY));
+                final Node childNode;
```

Comment:
Why not just declare the variable as it is set? There's no clarity gained by declaring it outside the `try`.